src/App/Routes/HomePage/GithubSignin/index.tsx

```diff
@@ -1,40 +1,40 @@
-import { useStarknet } from "@starknet-react/core";
-import { getStarknet } from "get-starknet";
 import { FC, useCallback, useEffect, useState } from "react";
 import { usePrevious } from "react-use";
+import { useRecoilValue } from "recoil";
 
 import config from "src/config";
 import { useGithubAccount } from "src/hooks/github-account";
 import { signMessage } from "src/utils/wallet";
 import GithubSignin from "src/App/Routes/HomePage/GithubSignin/View";
+import { accountAddressAtom, accountAtom } from "src/state";
 
 type Props = {
   className?: string;
 };
 
 const GithubSigninContainer: FC<Props> = ({ className }) => {
-  const { account } = useStarknet();
-  const starknet = getStarknet();
+  const accountAddress = useRecoilValue(accountAddressAtom);
+  const account = useRecoilValue(accountAtom);
```

Comment:
What's the difference between these two?

Reply:
`accountAddress` is the wallet address (a string like 0x000000000...)
`account` is the object implementing `AccountInterface` from starknet, which allow to send transaction and sign contract. This object is injected by ArgentX or Braavos browser extension

Reply:
Ok, then can't we deduce the `accountAddress` from the `account` in a Recoil selector?

Reply:
Technically we could but there are not synchronized at the same time.
And we need the `accountAddress` to deduce the `account`